src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -228,7 +228,15 @@ module.exports = function(
   }
 
 
-  function markUpdate(workInProgress) {
+  function markUpdateIfNecessary(workInProgress) {
+    const current = workInProgress.alternate;
+    const instance = workInProgress.stateNode;
+    if (!current && typeof instance.componentDidMount !== 'function') {
+      return;
+    }
+    if (current && typeof instance.componentDidUpdate !== 'function') {
```

Comment:
How would you combine them? It feels a bit messy to me to have `||` and `&&` with `!` in a single condition.